Aetherium/Artifacts/ArtifactOfRegression.cs

```diff
@@ -184,5 +199,108 @@ private void RegressAIToLowerForm(On.RoR2.CharacterAI.BaseAI.orig_OnBodyDeath or
             }
             orig(self, characterBody);
         }
+
+        private bool IsExisting(CharacterMaster master)
+        {
+            return master && master.IsDeadAndOutOfLivesServer();
+        }
+
+        private bool IsAnEnemy(CharacterMaster master)
+        {
+            return master.teamIndex == TeamIndex.Monster || master.teamIndex == TeamIndex.Lunar;
+        }
+
+        private void LogRegisteredRegressions()
+        {
+            ModLogger.LogMessage("Artifact of Regression Lookup Table:");
+            foreach (var pair in RegressionLookup)
+            {
+                ModLogger.LogMessage($"-> {pair.Key}");
+                foreach (var childPair in pair.Value.children)
+                {
+                    ModLogger.LogMessage($"  -> {childPair.Value.count} {childPair.Key}");
+                }
+            }
+        }
+    }
+
+    /// <summary>
+    /// A data structure useful for storing children data of a parent and registering them.
+    /// </summary>
+    public struct RegressData
+    {
+        private readonly string parentName;
+        internal Dictionary<string, ChildData> children;
```

Comment:
why does ChildData not just have childName as a field instead of making this a dictionary? pretty sure a list would be fine for this???

Reply:
I'm not sure why I haven't thought of that. It's probably because earlier I had plans for it and later disappeared.

Reply:
I'll check again to see after I sleep.

Reply:
Done.